contrib/rtlbrowse/logfile.c

```diff
@@ -93,1390 +89,1308 @@ static GtkTextTag *blue_tag = NULL, *fwht_tag = NULL;
-	if(found)
-		{
-		gchar cached = *found;
-		*found = 0;
-		offset = g_utf8_strlen (line_text, -1);
-		*found = cached;
-		break;
-		}
-	g_free (line_text);
+    for (;;) {
+        next = start;
+        gtk_text_iter_forward_line(&next);
+
+        /* No more text in buffer */
+        if (gtk_text_iter_equal(&start, &next)) {
+            free(strcaseins);
+            return (FALSE);
+        }
 
-	start = next;
-	}
+        pnt = line_text = gtk_text_iter_get_visible_text(&start, &next);
+        while (*pnt) {
+            *pnt = toupper((int)(unsigned char)*pnt);
+            pnt++;
+        }
+        found = strstr(line_text, strcaseins);
+        if (found) {
+            gchar cached = *found;
+            *found = 0;
+            offset = g_utf8_strlen(line_text, -1);
+            *found = cached;
+            break;
+        }
+        g_free(line_text);
 
-*match_start = start;
-forward_chars_with_skipping (match_start, offset);
+        start = next;
+    }
 
-offset = g_utf8_strlen (str, -1);
-*match_end = *match_start;
-forward_chars_with_skipping (match_end, offset);
+    *match_start = start;
+    forward_chars_with_skipping(match_start, offset);
 
-free(strcaseins);
-return(TRUE);
-}
+    offset = g_utf8_strlen(str, -1);
+    *match_end = *match_start;
+    forward_chars_with_skipping(match_end, offset);
 
+    free(strcaseins);
+    return (TRUE);
+}
 
 void tr_search_forward(char *str, gboolean noskip)
 {
-struct text_find_t *tr = selected_text_via_tab;
-
-if((tr) && (tr->text))
-	{
-	GtkTextBuffer *tb = gtk_text_view_get_buffer(GTK_TEXT_VIEW(tr->text));
-	GtkTextMark *tm = gtk_text_buffer_get_insert(tb);
-	GtkTextIter iter;
-	gboolean found = FALSE;
-	GtkTextIter match_start;
-	GtkTextIter match_end;
-
-	gtk_text_buffer_get_iter_at_mark(tb, &iter, tm);
-
-	tr->line = gtk_text_iter_get_line(&iter);
-	tr->offs = gtk_text_iter_get_line_offset(&iter);
-
-	if(!noskip)
-	if((tr->line == tr->srch_line) && (tr->offs == tr->srch_offs))
-		{
-		gtk_text_iter_forward_char(&iter);
-		}
-
-
-	if(str)
-		{
-		if(!matchcase_active)
-			{
-			found = iter_forward_search_caseins(&iter, str, &match_start, &match_end);
-			}
-			else
-			{
-			found = gtk_text_iter_forward_search(&iter, str, GTK_TEXT_SEARCH_TEXT_ONLY, &match_start, &match_end, NULL);
-			}
-		if(!found)
-			{
-			gtk_text_buffer_get_start_iter(tb, &iter);
-			if(!matchcase_active)
-				{
-				found = iter_forward_search_caseins(&iter, str, &match_start, &match_end);
-				}
-				else
-				{
-				found = gtk_text_iter_forward_search(&iter, str, GTK_TEXT_SEARCH_TEXT_ONLY, &match_start, &match_end, NULL);
-				}
-			}
-		}
-
-	if(found)
-		{
-		gtk_text_buffer_select_range(tb, &match_start, &match_end);
-		read_insert_position(tr);
-		tr->srch_line = tr->line;
-		tr->srch_offs = tr->offs;
-
-		/* tm = gtk_text_buffer_get_insert(tb); */ /* scan-build : never read */
+    struct text_find_t *tr = selected_text_via_tab;
+
+    if ((tr) && (tr->text)) {
```

Comment:
These kind of checks are common in the GTKWave code base. By negating them it is possible to reduce the nesting depth and make the code more readable:
```suggestion
    if (tr == NULL || tr->text == NULL) {
        return;
    }
```

Reply:
solved in 76abb3c2